store/v2/storage/sqlite/db.go

```diff
@@ -263,58 +249,23 @@ func (db *Database) ReverseIterator(storeKey []byte, version uint64, start, end
 }
 
 func (db *Database) PruneStoreKeys(storeKeys []string, version uint64) (err error) {
-	tx, err := db.storage.Begin()
-	if err != nil {
-		return fmt.Errorf("failed to create SQL transaction: %w", err)
-	}
-
-	defer func() {
-		err = tx.Rollback()
-	}()
-
 	// flush removed store keys
-	flushRemovedStoreKeysStmt := `
-	INSERT INTO state_storage(store_key, key, value, version)
-		VALUES(?, ?, ?, ?);
-	`
+	flushRemovedStoreKeysStmt := fmt.Sprintf(`INSERT INTO state_storage(store_key, key, value, version) 
+		VALUES %s`, strings.Repeat("(?, ?, ?, ?),", len(storeKeys)))
```

Comment:
🤔 I have never seen an insert with repeated placeholders. What are the benefits? With dynamic values possible, the DB can not optimize internally for the prepared statement AFAIK.
You also need to check for empty storeKeys slice or the statement would be incorrect.

The old version looks more common and trustworthy ;-)